src/inmanta/data/__init__.py

```diff
@@ -213,6 +233,296 @@ def ensure_boundary_type(
         else:
             return order_by_column_value
 
+    @property
+    def id_column(self) -> ColumnNameStr:
+        """Name of the id column of this database order"""
+        return ColumnNameStr("id")
+
+    def as_filter(
+        self,
+        offset: int,
+        column_value: Optional[object] = None,
+        id_value: Optional[Union[uuid.UUID, str]] = None,
+        start: Optional[bool] = True,
+    ) -> Tuple[List[str], List[object]]:
+        """ Get the column and id values as filters"""
+        filter_statements = []
+        values: List[object] = []
+        relation = ">" if start else "<"
+        if column_value is not None and id_value:
+            filter_statements.append(
+                f"({self.get_order_by_db_column_name()}, {self.id_column}) {relation} (${str(offset)}, ${str(offset + 1)})"
+            )
+            values.append(BaseDocument._get_value(column_value))
+            values.append(BaseDocument._get_value(id_value))
+        elif column_value is not None:
+            filter_statements.append(f"{self.get_order_by_db_column_name()} {relation} ${str(offset)}")
+            values.append(BaseDocument._get_value(column_value))
+        return filter_statements, values
+
+    def as_start_filter(
+        self,
+        offset: int,
+        start: Optional[object] = None,
+        first_id: Optional[Union[uuid.UUID, str]] = None,
+    ) -> Tuple[List[str], List[object]]:
+        """ Get the start and first_id values as start filters"""
+        return self.as_filter(offset, column_value=start, id_value=first_id, start=True)
+
+    def as_end_filter(
+        self,
+        offset: int,
+        end: Optional[object] = None,
+        last_id: Optional[Union[uuid.UUID, str]] = None,
+    ) -> Tuple[List[str], List[object]]:
+        """ Get the end and last_id values as end filters"""
+        return self.as_filter(offset, column_value=end, id_value=last_id, start=False)
+
+
+class ResourceOrder(DatabaseOrder):
+    """ Represents the ordering by which resources should be sorted"""
+
+    valid_sort_columns = {"resource_type": str, "agent": str, "status": str, "resource_id_value": str}
+
+    @classmethod
+    def validator_dataclass(cls) -> Type["BaseDocument"]:
+        return Resource
+
+    @classmethod
+    def parse_from_string(
+        cls,
+        sort: str,
+    ) -> "ResourceOrder":
+        database_order = super().parse_from_string(sort)
+        return ResourceOrder(database_order.order_by_column, database_order.order)
+
+    def get_order_by_db_column_name(self) -> ColumnNameStr:
```

Comment:
The columns sometimes have to be cast to a different type (for example casting an `enum` to `varchar` to sort alphabetically instead of the order of the enum), this method is used for this case, while in the API simply the name is used, since the use in the database is an implementation detail from that perspective. 
I renamed these methods to `get_order_by_column_db_name` and `get_order_by_column_api_name` respectively, to make this more explicit.

(This can be e.g. extracted to a separate class later)